File: Python/patten/cross_patten.py
# Print string of odd length in ‘X’ format
# Difficulty Level : Basic

if __name__ == "__main__":
    input_str = input('Enter the string value : ')
    if input_str:
        len_input_str = len(input_str)
        y = len_input_str -1
        for row in range(0,len_input_str):
            for col in range(0,len_input_str):
                if col == row or col == y:
                    print(input_str[col],end = " "),
                else:
                    print(" ",end = " "),
            y = y -1
            print("")
            

File: Javascript/patten/crossPatten.js
// Print string of odd length in ‘X’ format
// Difficulty Level : Basic

inputStr = 'test_data';
let x = inputStr.length
for(let i =0; i < inputStr.length;i++){
    for(let j=0; j< inputStr.length;j++){
        if(j == 1 || j ==x)
            document.write(str.charAt(k));
        else
            document.write(" ");
    }
    document.write("<br>");
    x=x-1;
} 